README.md

```diff
@@ -1,289 +1,590 @@
+  - name:  ACCOUNT_SERVICE_USERNAME
+    value: ((username))
+  - name:  ACCOUNT_SERVICE_PASSWORD
+    value: ((password))
+  - name:  ACCOUNT_SERVICE_URI
+    value: ((accountServiceUri))
 ```
 
-The recommended set of metadata will vary dependending on implementations and platforms, but two **RECOMMENDED** keys are:
-* kind
-* provider
+## Synthetic Provisioned Service
+
+There are some situations where there is an arity mismatch between a collection of Kubernetes resources representing a system and an application that wishes to consume them.  The solution to this problem is to create a synthetic Provisioned Service that is a composite of multiple other resources.
+
+A Synthetic Provisioned Service resource **MUST** define a `.spec.services` which is an array of `ObjectReference`s to a resource.
+
+A Synthetic Provisioned Service resource **MUST** define a `.spec.mappings` which is an array of `Mapping` objects. A `Mapping` object **MUST** define `name` and `value` entries. The value of a `Mapping` **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a [JSON Path](https://kubernetes.io/docs/reference/kubectl/jsonpath/) to an entry on a resource defined in `services`. The value of this `Secret` entry **MUST** be substituted into the original value string, replacing the token. Once all tokens have been substituted, the new value **MUST** be added to the binding `Secret` exposed by the resource.
 
+If a `.spec.kind` is set, the `kind` entry in the binding `Secret` **MUST** be set to its value. If a `.spec.provider` is set, the `provider` entry in the binding `Secret` **MUST** be set to its value.
 
-#### Exposing data as environment variables
+A Synthetic Provisioned Service resource **MUST** define a `.status.binding.name` which is a `LocalObjectReference` to a `Secret`. The `Secret` **MUST** be in the same namespace as the resource. The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding. It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding. The `Secret` **MAY** contain any other entry.
 
-The specification allows for binding properties to be additionally exposed as environment variables.  The assumption is that applications will have pre-knowledge of these environment variable keys, so the mechanisms described in this section are solely for the purpose of allowing the entity responsible for mounting the binding data to know which keys should also be made available as environment variables.  
+### Resource Type Schema
 
-This decision is made in the following ascending order of precedence:
-* value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
-* value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: SyntheticProvisionedService
```

Comment:
Hi @nebhale , why are we proposing a new resource to handle multiple provisioned services? 🤔

Reply:
I don't believe this is multiple services, but composing a single service from multiple upstream resources. The `.spec.services` key could use a better name.

Reply:
right. The scenario here is when we have data mappings that were created as a combination of data from multiple services.  When this is mounted it needs its own ID, so we're essentially creating a synthetic binding to host that composed data.

Reply:
I put a comment earlier in the PR about .spec.services.  We should really keep the notion of multiple services in the core `ServiceBinding` CR (even if we pick a better name for the key)

Reply:
My argument for this is that including this in the "core" specification (specifically in `ServiceBinding` where it existed before) adds complexity to the specification and makes interoperable implementations much more difficult to produce.  Because of the benefits of duck typing, any platform and any type can do things like this, e.g. synthesizing a new Provisioned Service from other concrete service resources.  Think of this as just another layer on the onion of abstraction.

That being said, as I was finishing up and writing the [commit message](https://github.com/application-stacks/service-binding-specification/pull/48/commits/c4c6ccb8b1b509922f68c3d7381b675b8bb16b1b) for this, I realized that I'm not sure that this should be specified.  I think letting platforms introduce any indirection they want, rather than trying to standardize what is a *much* more complex mapping syntax than the other extension, is a better way to go.

Reply:
Got it. Let me think this through a bit.

Reply:
To avoid getting into the discussion about one or more CRDs, (and there are good reasons for both suggestions), should we avoid having this CRD in the specification? The implementation may/may not need the second CRD .. and therefore not having it should not make it look non-spec-compliant.